scripts/checks/test_verify_trivy_pr_scan.py

```
@@ -0,0 +1,594 @@
+    "explicit pull-request eligibility conditions": EXPLICIT_PR_ELIGIBILITY,
+    "recursive PR-eligible needs chain": PR_ELIGIBLE_NEEDS_CHAIN,
+}
+
+
+def _run_checker(workflow_text: str) -> subprocess.CompletedProcess[str]:
+    """Run the production checker against one isolated workflow fixture."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        workflow_path = Path(temp_dir) / ".github" / "workflows" / "trivy.yml"
+        workflow_path.parent.mkdir(parents=True)
+        workflow_path.write_text(workflow_text, encoding="utf-8")
+        return subprocess.run(
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)